final/tool/menus.cpp

```diff
@@ -0,0 +1,192 @@
+#include "stdafx.h"
+#include <ostream>
+#include <iostream>
+#include <string>
+#include <sstream>
+#include <fstream>
+#include <map>
+#include <vector>
+#include "block.h"
+#include <algorithm>
+#include "menus.h"
+
+void press_enter_to_go_back()
+{
+  std::cout << "\nPress ENTER to go back...\n";
+  std::cin.ignore();
+  system("cls");
+}
+
+unsigned int get_number_from_console(std::string caption)
+{
+  unsigned int user_number = 0;
+  std::string input = "";
+  while (true) {
+    std::cout << caption;
+    getline(std::cin, input);
+    // This code converts from string to number safely.
+    std::stringstream input_ss(input);
+    if (input_ss >> user_number)
+      break;
+    std::cout << "Invalid number, please try again..." << std::endl;
+  }
+  return user_number;
+}
+
+double get_input_from_console()
+{
+  double input_value = 0.;
+  std::string input;
+  while (true) {
+    std::cout << "Please enter a valid input number: ";
+    getline(std::cin, input);
+    // This code converts from string to number safely.
+    std::stringstream input_ss(input);
+    if (input_ss >> input_value)
+      break;
+    std::cout << "Invalid input value, please try again..." << std::endl;
+  }
+  return input_value;
+}
+
+void show_available_block_types()
+{
+  //show available block types
+  system("cls");
+  std::string block_list_menu = "Available block types:\n";
+  std::vector<std::string> block_list = block_factory::GetBlockTypes();
+  for (std::vector<std::string>::iterator it = block_list.begin(); it != block_list.end(); ++it)
+    block_list_menu += *it;
+  std::cout << block_list_menu << std::endl;
+}
+
+void show_block_sequence(block_processor &bp)
+{
+  //show block sequence
+  system("cls");
+  std::cout << "Block sequence:\n" << std::endl;
+  int index = 0;
+  std::vector<std::string> temp_v = bp.get_sequence();
+  std::for_each(temp_v.begin(), temp_v.end(), [&](std::string const &c)

```

Comment:
ili samo obican for kada je vec potreban index?

Reply:
see above